internal/objects/crdrules.go

```diff
@@ -178,12 +239,37 @@ func (c *CRDLister) UpdateFqdnHTTPRulesMappings(fqdn, path, httprule string) {
 }
 
 // FqdnSharedVSModelCache/SharedVSModelFqdnCache
-func (c *CRDLister) GetFQDNToSharedVSModelMapping(fqdn string) (bool, string) {
-	found, modelName := c.FqdnSharedVSModelCache.Get(fqdn)
-	if !found {
-		return false, ""
+func (c *CRDLister) GetFQDNToSharedVSModelMapping(fqdn string) (bool, []string) {
+	c.NSLock.Lock()
+	defer c.NSLock.Unlock()
+	oktype, fqdnType := c.FqdnFqdnTypeCache.Get(fqdn)
+	if !oktype || fqdnType == "" {
+		fqdnType = string(akov1alpha1.Exact)
 	}
-	return true, modelName.(string)
+
+	allFqdns := c.FqdnSharedVSModelCache.GetAllKeys()
```

Comment:
it seems you are nesting a lock here. The GetAllKeys already has a READ lock inside of it. Is this intentional?